components/mediation/tasks/org.wso2.micro.integrator.ntask.core/src/main/java/org/wso2/micro/integrator/ntask/core/impl/FileBasedTaskRepository.java

```diff
@@ -332,4 +339,12 @@ public String getTaskMetadataProp(String taskName, String key) throws TaskExcept
                                     TaskException.Code.UNKNOWN, e);
         }
     }
+
+    private static XMLStreamReader getXMLStreamReader(InputStream input) throws XMLStreamException {
+        XMLInputFactory xmlInputFactory = XMLInputFactory.newFactory();
```

Comment:
Shall we instantiate this at the class level? AFAIK, this takes considerable time to get initiated.  Similar to https://github.com/wso2/wso2-synapse/pull/1594

Reply:
Fixed with 2b88edc